WalletWasabi/Extensions/RPCClientExtensions.cs

```diff
@@ -77,9 +77,33 @@ public static async Task<AllFeeEstimate> EstimateAllFeeAsync(this IRPCClient rpc
 			var mempoolInfo = await rpc.GetMempoolInfoAsync().ConfigureAwait(false);
 			var sanityFeeRate = mempoolInfo.GetSanityFeeRate();
 
+			const int BlockSize = 1_000_000;
+			static IEnumerable<(int Size, FeeRate From, FeeRate To)> SplitFeeGroupInBlocks(FeeRateGroup group)
+				=> Enumerable
+					.Range(0, (int)group.Sizes / BlockSize)
+					.Select(b => (Sizes: BlockSize, From: group.From, To: group.To));
+
+			var minFeeRatePerTarget = mempoolInfo.Histogram
```

Comment:
This is quite hardcore code to actually verify.

Reply:
Yes, it is. I invested a lot of brain power trying to make it as short as possible but it does too much. Any suggestion? Do you think I should break it in smaller expressions?

Reply:
I've refactored (splitted across multiple lines with comments). Please take a look.